### PR TITLE
Gun switcher

### PR DESCRIPTION
**Pull request by matei-dorian**
Personajul poate schimba armele folosind rotita mouseului.
Pentru a obtine acest efect am great toate armele ca game objects si le am grupat intr un empty object. La acel empty object am adaugat un script care itereaza prin obiectele copil si da disable/enable la celelalte arme in functie de ce arma este selectata.

Animatiile nu merg inca, dar vom incerca in sprintul viitor sa reparam acest bug.

### CODE REVIEW

File: Game Dev/Assets/Scripts/Player/WeaponSwitch.cs [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        SelectWeapon();
+    }
+
+    void Update()
+    {
+        if (Input.GetAxis("Mouse ScrollWheel") > 0f)
+            SelectedWeapon = (SelectedWeapon + 1) % transform.childCount;
+        if (Input.GetAxis("Mouse ScrollWheel") < 0f)
+            SelectedWeapon = (SelectedWeapon + transform.childCount - 1) % transform.childCount;
+        SelectWeapon();
```
Comment (NMDMaria) [inline Game Dev/Assets/Scripts/Player/WeaponSwitch.cs:19]: No need to call switch at each frame, would be better if only called when the SelectedWeapon is changed.

**Review by NMDMaria — CHANGES_REQUESTED** [review]
Firepoint should match each weapon. Because of the change, gameobject WeaponShoot is missing so shooting animations can't roll.

File: Game Dev/Assets/Scripts/Player/WeaponSwitch.cs (hunk 1 above)
Comment (matei-dorian) [reply Game Dev/Assets/Scripts/Player/WeaponSwitch.cs:19]: There is no need to be this precise, SelectWeapon() only iterates through an array of 3 elements so the time complexity is negligible even if we call the function on each frame or on each Mouse ScrollWheel.

**Comment by matei-dorian** [comment]
> Firepoint should match each weapon. Because of the change, gameobject WeaponShoot is missing so shooting animations can't roll.

We will add this as a bug in Jira for the next sprint

**Pull request merged**